cli/dcoscli/data/help/package.txt

```diff
@@ -21,7 +21,7 @@ Usage:
     dcos package repo remove <repo-names>...
     dcos package search [<query> --json]
     dcos package uninstall <package-name>
-                           [--cli | [--app --app-id=<app-id> --all --yes]]
+                           [--cli --app --app-id=<app-id> --all --yes]
```

Comment:
I don't know how to do it off the top of my head, but it would be great if we could force docopt to recognize that `--app-id` and `--all` are only valid if you also pass `--app`.

Reply:
This is not the case, for example `dcos package uninstall hello-world --all` is valid.

Actually, I'm not convinced about our change here (to allow both `--cli` and `--app`) : 
- in terms of semantics, those options are described as `CLI only` or `application only` (https://github.com/bamarni/dcos-cli/blob/02444d2cc9d7619f22eecadf7ef710aa67977d2b/cli/dcoscli/data/help/package.txt#L48-L53), so it's kind of strange to use both.
- it doesn't align with `dcos package install` command, which would still have them as mutually exclusive (https://github.com/bamarni/dcos-cli/blob/02444d2cc9d7619f22eecadf7ef710aa67977d2b/cli/dcoscli/data/help/package.txt#L48-L53).

Reply:
Or I think you meant : `--app-id` and `--all` are not valid if you pass `--cli`?

Reply:
Yeah, OK, that makes sense. I'm OK going back to the old semantics if the docs (and other functions) are explicit about the ONLY part (meaning put the old docopt string back in place). That shouldn't change anything about the new code flow I suggested though (other than to mention in the comments that only one of cli or app will ever be true at a given time).